src/test/scala/elevate/rise/tvmGemm.scala

```diff
@@ -0,0 +1,193 @@
+  val vectorization: Strategy[Rise] =
+    blocking `;;`
+      (vectorize(32) `@` innermost(isApplied(isApplied(isMap))))
+
+  test("vectorization") {
+    run("vectorization", (vectorization `;` lowerToC), openMP = true)
+  }
+
+  // -- LOOP PERMUTATION -------------------------------------------------------
+
+  val loopPerm: Strategy[Rise] = baseline `;`
+    (tile(32,32)        `@` outermost(mapNest(2))) `;;`
+    (reduceMapFission() `@` outermost(isApplied(isApplied(isReduceSeq)))) `;;`
+    (splitStrategy(4)   `@` innermost(isFullyAppliedReduce)) `;;`
+    reorder(List(1,2,5,3,6,4)) `;;`
+    (vectorize(32) `@` innermost(isFullyAppliedMap))
+
+  test("loop permutation") {
+    run("loop_permutation", (loopPerm `;` lowerToC), openMP = true)
+  }
+
+  // -- ARRAY PACKING ----------------------------------------------------------
+
+  val isTransposedB: Strategy[Rise] = isApplied(isTranspose)
+  val permuteB: Strategy[Rise] =
+      splitJoin2(32) `;` DFNF() `;` argument(idAfter) `;`
+      topDown(liftId) `;` topDown(createTransposePair) `;` RNF() `;`
+      argument(argument(idAfter)) `;` normalize.apply(liftId) `;`
+      topDown(idToCopy)
+
+  val packB: Strategy[Rise] =
+    storeInMemory(isTransposedB,
+      permuteB `;;`
+        (vectorize(32) `@` innermost(isFullyAppliedMap)) `;;`
+        (parallel()    `@` outermost(isApplied(isMap)))
+    ) `@` inLambda
+
+  def inLambda(s: Strategy[Rise]): Strategy[Rise] =
+    isLambda `;` ( (e: Rise) => body(inLambda(s))(e) ) <+ s
+
+  val arrayPacking: Strategy[Rise] = packB `;;` loopPerm
+  test("array packing") {
+    run("array_packing", arrayPacking `;` lowerToC, openMP = true)
+  }
+
+  // -- CACHE BLOCKS -----------------------------------------------------------
+
+  val cacheBlocks: Strategy[Rise] = (
+    arrayPacking `;;` debug[Rise]("after arrayPacking") `;`
+      (unroll `@` innermost(isReduceSeq))
+    )
+
+  test("cache blocks") {
+    run("cache_blocks", cacheBlocks `;` lowerToC, openMP = true)
+  }
+
+  // -- PARALLEL ---------------------------------------------------------------
+
+  val par = (
+    arrayPacking `;;`
+      ((parallel() `@` outermost(isApplied(isMap))) `@`
+        outermost(isApplied(isLet))) `;;`
+      (unroll `@` innermost(isReduceSeq))
+    )
+
+  test("parallel") {
+    run("parallel", par `;` lowerToC, openMP = true)
+  }
+
+
+  /// UTILS ////////////////////////////////////////////////////////////////////
+
+//  val kernelsFolder: String = "/home/artifact/kernels"
+//  val plotsFolder: String = "/home/artifact/results/fig10/steps"
+
+  def run(version: String,
+          strategy: Strategy[Rise],
+          openMP: Boolean // generate C or OpenMP code?
+         ): Unit = {
+
+//    def toFile(path: String, name: String, content: String, ending: String = ".c"): Unit = {
+//      import java.io._
+//      val w =new PrintWriter(new File(s"$path/$name$ending"))
+//      w.write(content)
+//      w.flush()
+//      w.close()
+//    }
```

Comment:
do we need to keep these commented out code chunks?

Reply:
I have uncommented the code and made it an easy to opt-in option of the test to store the intermediate results on disc (as it was done for the ICFP 2020 artifact).